src/systems/systems/src/navigation/radio_altimeter.rs

```diff
@@ -0,0 +1,864 @@
+        self.pitch = Angle::new::<degree>(reader.read(&self.pitch_id));
+        self.bank = Angle::new::<degree>(reader.read(&self.bank_id));
+    }
+}
+
+pub trait RadioAltimeter {
+    fn radio_altitude(&self) -> Arinc429Word<Length>;
+}
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum Ala52BAircraftInstallationDelay {
+    FortyFeet,
+    FiftySevenFeet,
+    EightyFeet,
+}
+
+impl Ala52BAircraftInstallationDelay {
+    fn delay(&self) -> Length {
+        match self {
+            Ala52BAircraftInstallationDelay::FortyFeet => Length::new::<foot>(40.),
+            Ala52BAircraftInstallationDelay::FiftySevenFeet => Length::new::<foot>(57.),
+            Ala52BAircraftInstallationDelay::EightyFeet => Length::new::<foot>(80.),
+        }
+    }
+}
+
+struct Ala52BRadioAltimeterRuntime {
+    remaining_self_check: Duration,
+    is_tracking_ground: bool,
+    stale_for: Duration,
+    aircraft_installation_delay: Ala52BAircraftInstallationDelay,
+    radio_altitude: Length,
+}
+
+impl Ala52BRadioAltimeterRuntime {
+    const ARINC_429_REFRESH_INTERVAL: Duration = Duration::from_millis(80);
+
+    fn new(self_check_time: Duration, aid: Ala52BAircraftInstallationDelay) -> Self {
+        Self {
+            remaining_self_check: self_check_time,
+            is_tracking_ground: false,
+            stale_for: Self::ARINC_429_REFRESH_INTERVAL,
+            aircraft_installation_delay: aid,
+            radio_altitude: Length::new::<foot>(0.),
+        }
+    }
+
+    fn new_running(aid: Ala52BAircraftInstallationDelay) -> Self {
+        Self {
+            remaining_self_check: Duration::ZERO,
+            is_tracking_ground: false,
+            stale_for: Self::ARINC_429_REFRESH_INTERVAL,
+            aircraft_installation_delay: aid,
+            radio_altitude: Length::new::<foot>(0.),
+        }
+    }
+
+    fn update(&mut self, context: &UpdateContext, transceivers: &impl TransceiverPair) {
+        // First, check if we're still starting up and if so, simulate a wait until all self tests
+        // have completed
+        if let Some(new_remaining) = self.remaining_self_check.checked_sub(context.delta()) {
+            self.remaining_self_check = new_remaining;
+        } else {
+            self.remaining_self_check = Duration::ZERO;
+        }
+
+        // If there's any startup time remaining, do nothing
+        if self.remaining_self_check > Duration::ZERO {
+            return;
+        }
+
+        // Update only once per refersh interval
+        if self.stale_for >= Self::ARINC_429_REFRESH_INTERVAL {
+            // Interrogate the transceivers for a response (which might fail if no physical path exists
+            // or some other kind of failure).
+            if let Some(response) = transceivers.response() {
+                let travel_time = response.travel_time();
+
+                let speed: Velocity = Velocity::new::<meter_per_second>(299_792_458.)
```

Comment:
**suggestion**: Consider adding a const to the module and using it in both places where you define it:
```rust
const SPEED_OF_LIGHT_METER_PER_SECOND: f64 = 299_792_458.;
```